libpod/container_internal.go

```diff
@@ -2738,3 +2739,122 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 
 	return nil
 }
+
+func (c *Container) resetHealthCheckTimers(noHealthCheck bool, changedTimer bool, isStartup bool) error {
+	if !c.ensureState(define.ContainerStateCreated, define.ContainerStateRunning) {
+		return nil
+	}
+
+	switch {
+	case noHealthCheck:
+		if err := c.removeTransientFiles(context.Background(),
+			c.config.StartupHealthCheckConfig != nil && !c.state.StartupHCPassed,
+			c.state.HCUnitName); err != nil {
+			return err
+		}
+	case isStartup && changedTimer && c.config.StartupHealthCheckConfig != nil && !c.state.StartupHCPassed:
+		c.state.StartupHCPassed = false
+		c.state.StartupHCSuccessCount = 0
+		c.state.StartupHCFailureCount = 0
+		if err := c.save(); err != nil {
+			return err
+		}
+		c.recreateHealthCheckTimer(context.Background(), true, true)
+	case changedTimer && c.config.HealthCheckConfig != nil:
+		c.recreateHealthCheckTimer(context.Background(), false, false)
+	}
+	return nil
+}
+
+func (c *Container) updateHealthCheckConfiguration(healthCheckConfig *manifest.Schema2HealthConfig, changedTimer bool, noHealthCheck bool) error {
+	oldHealthCheck := &manifest.Schema2HealthConfig{}
+	if c.config.HealthCheckConfig != nil {
+		if err := JSONDeepCopy(c.config.HealthCheckConfig, oldHealthCheck); err != nil {
+			return err
+		}
+	}
+
+	c.config.HealthCheckConfig = healthCheckConfig
+
+	if err := c.runtime.state.SafeRewriteContainerConfig(c, "", "", c.config); err != nil {
+		// Assume DB write failed, revert to old resources block
+		c.config.HealthCheckConfig = oldHealthCheck
+		return err
+	}
+
+	err := c.resetHealthCheckTimers(noHealthCheck, changedTimer, false)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debugf("HealthCheck configuration updated for container %s", c.ID())
+	c.newContainerEvent(events.UpdateHealthCheckConfig)
+	return nil
+}
+
+func (c *Container) updateStartupHealthCheckConfiguration(startupHealthCheckConfig *define.StartupHealthCheck, changedStartupTimer bool, noHealthCheck bool) error {
```

Comment:
This is basically identical to updateHealthCheckConfiguration minus a few variables; can the two be refactored to share code? It seems like everything from line 2805 on is basically identical.